mlir/include/mlir/Dialect/ArmSVE/IR/ArmSVE.td

```diff
@@ -410,4 +411,24 @@ def ConvertToSvboolIntrOp :
     /*overloadedResults=*/[]>,
     Arguments<(ins SVEPredicate:$mask)>;
 
+// Note: This multi-vector intrinsic requires SME2.
+def ZipX2IntrOp : ArmSVE_IntrOp<"zip.x2",
+    /*traits=*/[],
```

Comment:
AllTypesMatch? (and for x4)

Reply:
All the SVE intrinsics are super lax like this. '`AnyScalableVector`' is much more than these can actually take, I've left the type checking to the actual ops.